#!/usr/bin/env python3
PACKAGE = "nextage_optas"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("x_min", double_t, 0, "Workspace min boundary in x in meters", 0.0, -1.0, 1.0)
gen.add("x_max", double_t, 0, "Workspace max boundary in x in meters", 0.0, -1.0, 1.0)
gen.add("y_min", double_t, 0, "Workspace min boundary in y in meters", 0.0, -1.0, 1.0)
gen.add("y_max", double_t, 0, "Workspace max boundary in y in meters", 0.0, -1.0, 1.0)
gen.add("z_min", double_t, 0, "Workspace min boundary in z in meters", 0.0, -1.0, 1.0)
gen.add("z_max", double_t, 0, "Workspace max boundary in z in meters", 0.0, -1.0, 1.0)
gen.add("roll_min", double_t, 0, "Workspace min boundary in roll angle in degrees", 0.0, -360, 360)
gen.add("roll_max", double_t, 0, "Workspace max boundary in roll angle in degrees", 0.0, -360, 360)
gen.add("pitch_min", double_t, 0, "Workspace min boundary in pitch angle in degrees", 0.0, -360, 360)
gen.add("pitch_max", double_t, 0, "Workspace max boundary in pitch angle in degrees", 0.0, -360, 360)
gen.add("yaw_min", double_t, 0, "Workspace min boundary in yaw angle in degrees", 0.0, -360, 360)
gen.add("yaw_max", double_t, 0, "Workspace max boundary in yaw angle in degrees", 0.0, -360, 360)
gen.add("W_x", double_t, 0, "Cost gain for task-space in x", 0.0, 0, 50)
gen.add("W_y", double_t, 0, "Cost gain for task-space in y", 0.0, 0, 50)
gen.add("W_z", double_t, 0, "Cost gain for task-space in z", 0.0, 0, 50)
gen.add("W_roll", double_t, 0, "Cost gain for task-space in roll", 0.0, 0, 5)
gen.add("W_pitch", double_t, 0, "Cost gain for task-space in pitch", 0.0, 0, 5)
gen.add("W_yaw", double_t, 0, "Cost gain for task-space in yaw", 0.0, 0, 5)

exit(gen.generate(PACKAGE, "nextage_optas", "CmdTwistSingleArm"))